source/brailleDisplayDrivers/seikantk.py

```diff
@@ -57,66 +59,121 @@
 SEIKA_KEYS = b"\xff\xff\xa6"
 SEIKA_KEYS_ROU = b"\xff\xff\xa8"
 
-SEIKA_CONFIG = b"\x50\x00\x00\x25\x80\x00\x00\x03\x00"
+BAUD = 9600
+SEIKA_HID_FEATURES = b"".join([
+	b"\x50\x00\x00",
+	int.to_bytes(BAUD, length=2, byteorder="big", signed=False),  # b"\x25\x80"
+	b"\x00\x00\x03\x00",
+])
 SEIKA_CMD_ON = b"\x41\x01"
 
 vidpid = "VID_10C4&PID_EA80"
 hidvidpid = "HID\\VID_10C4&PID_EA80"
 SEIKA_NAME = "seikantk"
 
+# Bluetooth name of the Seika devices is "TSM abcd", where the "abcd" is a four-digit
+# number, e.g. "TSM 3366", "TSM 0001", etc. There is a space between "TSM" and "abcd".
+seikaBluetoothNameRegex = re.compile(r"TSM \d\d\d\d")
+
+
+def isSeikaBluetoothName(bluetoothName: str) -> bool:
+	return bool(seikaBluetoothNameRegex.match(bluetoothName))
+
+
+def isSeikaBluetoothDeviceInfo(deviceInfo: typing.Dict[str, str]) -> bool:
+	return isSeikaBluetoothName(deviceInfo["bluetoothName"])
+
+
+def isSeikaBluetoothDeviceMatch(match: DeviceMatch) -> bool:
+	return isSeikaBluetoothDeviceInfo(match.deviceInfo)
+
 
 class BrailleDisplayDriver(braille.BrailleDisplayDriver):
 	_dev: hwIo.IoBase
 	name = SEIKA_NAME
 	# Translators: Name of a braille display.
 	description = _("Seika Notetaker")
-	path = ""
 	isThreadSafe = True
-	for d in hwPortUtils.listHidDevices():
-		if d["hardwareID"].startswith(hidvidpid):
-			path = d["devicePath"]
 
 	@classmethod
 	def getManualPorts(cls) -> typing.Iterator[typing.Tuple[str, str]]:
 		"""@return: An iterator containing the name and description for each port.
 		"""
-		return braille.getSerialPorts()
+		return braille.getSerialPorts(isSeikaBluetoothDeviceInfo)
 
-	def __init__(self, port="hid"):
+	def __init__(self, port=bdDetect.KEY_HID):
```

Comment:
While I'm not very familiar with NVDA's braille display drivers, this is the only one which specified `HID` as the value for the `port` keyword argument in the constructor. For all others we either leave it with no default value in case of displays where port needs to be specified by the user, or for displays which can be auto-detected we assign `"auto"` to it. Could you add a comment here to explain why this driver is different?

Reply:
The short answer is I was trying to avoid making changes if not necessary. Looking at this closer, it seems like it should never be "hid". I have tried to improve the docs and typing elsewhere. One confusing thing in the NVDA braille code is that "port" is used to refer to many different things, in different parts of the code it may be a COM port string, a string description of where to look (auto, usb, bluetooth), a HID device path string, or a DeviceMatch class instance.